src/lib/RoutingManager.ts

```diff
@@ -34,17 +34,13 @@ class RoutingManager {
     this.instantSearchInstance = instantSearchInstance;
 
     this.createURL = this.createURL.bind(this);
-    this.applyStateFromRoute = this.applyStateFromRoute.bind(this);
-
-    this.router.onUpdate(this.applyStateFromRoute);
+    this.applySearchParameters = this.applySearchParameters.bind(this);
   }
 
-  public applyStateFromRoute(route: UiState): void {
-    const currentUiState = this.stateMapping.routeToState(route);
-
+  public applySearchParameters(uiState: UiState): void {
```

Comment:
I've renamed the function to a broader name, we already have the type for more information.